### PR TITLE
Close: Feat delete item

### PR DESCRIPTION
**Pull request by Sgalvin36**
## Scope

Added test to items_request_spec to test the route for destroying a record off the item table
Added the route to the routes file
Added the action to the item_controller and incorporated returning the correct status update and making sure that no body is given with that status.

### CODE REVIEW

**Review by jdavalos98 — APPROVED** [review]
code is looking dry and your route is RESTful. Thank you for the organizing of our routes file.

File: spec/requests/v1/items_request_spec.rb [hunk 1]
```diff
@@ -93,4 +89,19 @@
...
+            price: 69.00,
+            merchant_id: @merchant.id)
+        
+        expect(Item.count).to eq(5)
+    
+        delete "/api/v1/items/#{item.id}"
+    
+        expect(response).to be_successful
+        expect(Item.count).to eq(10)
+        expect{Item.find(poster.id) }.to raise_error(ActiveRecord::RecordNotFound)
```
Comment (kylomite) [inline spec/requests/v1/items_request_spec.rb]: This line tests to ensure the specified item was removed from items list using the`raise_error` matcher with `ActiveRecord::RecordNotFound`

File: spec/requests/v1/items_request_spec.rb [hunk 2]
```diff
@@ -93,4 +89,19 @@
...
+            description: "Hard work rarely pays off.",
+            price: 69.00,
+            merchant_id: @merchant.id)
+        
+        expect(Item.count).to eq(5)
+    
+        delete "/api/v1/items/#{item.id}"
+    
+        expect(response).to be_successful
+        expect(Item.count).to eq(10)
```
Comment (kylomite) [inline spec/requests/v1/items_request_spec.rb]: I'm not sure why the `Item.count` value goes from `5` before the delete, and becomes `10`.  I would like to talk about this in more detail, if you find time can you elaborate on this decision?

File: spec/requests/v1/items_request_spec.rb [hunk 3]
```diff
@@ -101,7 +101,7 @@
         delete "/api/v1/items/#{item.id}"
     
         expect(response).to be_successful
-        expect(Item.count).to eq(10)
-        expect{Item.find(poster.id) }.to raise_error(ActiveRecord::RecordNotFound)
+        expect(Item.count).to eq(4)
```
Comment (kylomite) [inline spec/requests/v1/items_request_spec.rb:104]: Scratch my previous comment. This was what I initially expected this value to become, good catch :)

File: spec/requests/v1/items_request_spec.rb (hunk 1 above)
Comment (kylomite) [reply spec/requests/v1/items_request_spec.rb]: this concern was addressed in the 4th commit, disregard comment.

**Review by kylomite — APPROVED** [review]

**Pull request merged**